lib/core/src/display/navigation/navigator.rs

```diff
@@ -95,25 +95,23 @@ impl Navigator {
 
         let dom_clone = dom.clone();
         let zoom_callback = move |zoom:ZoomEvent| {
-            if let CameraType::Perspective(persp) = camera.camera_type() {
                 let point      = zoom.focus;
                 let normalized = normalize_point2(point, dom_clone.dimensions());
                 let normalized = normalized_to_range2(normalized, -1.0, 1.0);
 
                 // Scale X and Y to compensate aspect and fov.
-                let x                     = -normalized.x * persp.aspect;
-                let y                     =  normalized.y;
-                let fov_slope             = camera.get_y_scale();
-                let z                     = fov_slope;
-                let direction             = Vector3::new(x, y, z).normalize();
-                let mut position          = properties.spring().fixed_point;
-                let zoom_amount           = zoom.amount * position.z;
-                position                 += direction   * zoom_amount;
-                let min_zoom              = persp.near  + min_zoom;
-                position.z                = clamp(position.z, min_zoom, max_zoom);
+                let x            = -normalized.x * camera.screen().aspect();
+                let y            =  normalized.y;
+                let fovy_slope   = camera.half_fovy_slope();
+                let z            = 1.0 / fovy_slope;
```

Comment:
It makes perfect sense. I would change here few things. Variable name `fovy_slope` -> `half_fovy_slope`. And `let y_unit = 1.0` :)